cmd/ack-generate/command/olm.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	generate "github.com/aws-controllers-k8s/code-generator/pkg/generate"
+	ackgenerate "github.com/aws-controllers-k8s/code-generator/pkg/generate/ack"
+	olmgenerate "github.com/aws-controllers-k8s/code-generator/pkg/generate/olm"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+)
+
+const (
+	olmConfigFileSuffix = "olmconfig.yaml"
+)
+
+// optOLMConfigPath
+var optOLMConfigPath string
+var optDisableCommonLinks bool
+var optDisableCommonKeywords bool
+
+// olmCmd is the command that generates a service ClusterServiceVersion base
+// for generating an operator lifecycle manager bundle.
+var olmCmd = &cobra.Command{
+	Use: "olm <service> <version>",
+	Short: `Generate an Operator Lifecycle Manager's ClusterServiceVersion
+resources for AWS service API. Expects a configuration file to be passed in
+via the --olm-config option, or otherwise reads the config at the root of the
+project with the filename <servicealias>-olmconfig.yaml`,
+	RunE: generateOLMAssets,
+}
+
+func init() {
+	olmCmd.PersistentFlags().StringVar(
+		&optOLMConfigPath, "olm-config", "", "the OLM configuration file to inform how OLM assets are generated.",
+	)
+	// optionally disable common links in the resulting manifest
+	olmCmd.PersistentFlags().BoolVar(
+		&optDisableCommonLinks, "no-common-links", false, "does not include common links in the rendered cluster service version.",
+	)
+	// optionally disable common keywords in the resulting manifest
+	olmCmd.PersistentFlags().BoolVar(
+		&optDisableCommonKeywords, "no-common-keywords", false, "does not include common keywords in the rendered cluster service version",
+	)
+
+	rootCmd.AddCommand(olmCmd)
+}
+
+// generateOLMAssets generates all assets necessary for delivering the
+// service controllers via operator lifecycle manager ("OLM").
+func generateOLMAssets(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return fmt.Errorf(
+			"please specify the service alias and version " +
+				"for the AWS service API to generate",
+		)
+	}
+	svcAlias := strings.ToLower(args[0])
+	if optOutputPath == "" {
+		optOutputPath = filepath.Join(optServicesDir, svcAlias)
+	}
+
+	version := args[1]
```

Comment:
After working with the `build-controller-release.sh` script, I'm wondering if I should accept "v0.0.1" instead of "0.0.0" here. The script accepts the leading `v`, this expects just the numbers. Small nitpick on my end. I generally prefer just the numbers but also want to keep things consistent.

Reply:
Up to you, @komish. We could also (long-term, not needed in this PR) use https://github.com/Masterminds/semver to handle this kind of common stuff.